addOns/automation/src/main/java/org/zaproxy/addon/automation/AuthenticationData.java

```diff
@@ -99,40 +109,55 @@ public AuthenticationData() {}
 
     public AuthenticationData(Context context) {
         AuthenticationMethod authMethod = context.getAuthenticationMethod();
-        if (authMethod instanceof HttpAuthenticationMethod) {
-            HttpAuthenticationMethod httpAuthMethod = (HttpAuthenticationMethod) authMethod;
+        if (authMethod instanceof HttpAuthenticationMethod httpAuthMethod) {
             setMethod(AuthenticationData.METHOD_HTTP);
             JobUtils.addPrivateField(parameters, PARAM_REALM, httpAuthMethod);
             JobUtils.addPrivateField(parameters, PARAM_HOSTNAME, httpAuthMethod);
             JobUtils.addPrivateField(parameters, PARAM_PORT, httpAuthMethod);
-        } else if (authMethod instanceof FormBasedAuthenticationMethod) {
-            FormBasedAuthenticationMethod formAuthMethod =
-                    (FormBasedAuthenticationMethod) authMethod;
+        } else if (authMethod instanceof FormBasedAuthenticationMethod formAuthMethod) {
             setMethod(AuthenticationData.METHOD_FORM);
             JobUtils.addPrivateField(parameters, PARAM_LOGIN_PAGE_URL, formAuthMethod);
             JobUtils.addPrivateField(
                     parameters, PARAM_LOGIN_REQUEST_URL, FIELD_LOGIN_REQUEST_URL, formAuthMethod);
             JobUtils.addPrivateField(parameters, PARAM_LOGIN_REQUEST_BODY, formAuthMethod);
-        } else if (authMethod instanceof JsonBasedAuthenticationMethod) {
-            JsonBasedAuthenticationMethod jsonAuthMethod =
-                    (JsonBasedAuthenticationMethod) authMethod;
+        } else if (authMethod instanceof JsonBasedAuthenticationMethod jsonAuthMethod) {
             setMethod(AuthenticationData.METHOD_JSON);
             JobUtils.addPrivateField(parameters, PARAM_LOGIN_PAGE_URL, jsonAuthMethod);
             JobUtils.addPrivateField(
                     parameters, PARAM_LOGIN_REQUEST_URL, FIELD_LOGIN_REQUEST_URL, jsonAuthMethod);
             JobUtils.addPrivateField(parameters, PARAM_LOGIN_REQUEST_BODY, jsonAuthMethod);
-        } else if (authMethod instanceof ScriptBasedAuthenticationMethod) {
-            ScriptBasedAuthenticationMethod scriptAuthMethod =
-                    (ScriptBasedAuthenticationMethod) authMethod;
-            ScriptWrapper sw = (ScriptWrapper) JobUtils.getPrivateField(scriptAuthMethod, "script");
+        } else if (authMethod != null
+                && authMethod
+                        .getClass()
+                        .getCanonicalName()
+                        .equals(CLIENT_SCRIPT_BASED_AUTH_METHOD_CLASSNAME)) {
+            // TODO: Plan to change once the core supports dynamic methods better
```

Comment:
This should be moved to the top or some common place, same as `BROWSER_BASED_AUTH_METHOD_CLASSNAME`.